### ISSUE TITLE
Corber: Unknown browser undefined

### ISSUE THREAD

**Issue by OakBehringer** [isleofcode/corber#509]
When trying to run corber serve...

```
Corber: Unknown browser undefined
Unknown browser undefined
```

Stack Trace and Error Report: 
```
ENV Summary:

  TIME: Mon Jul 02 2018 12:25:15 GMT-0700 (PDT)
  TITLE: corber
  ARGV:
  - /usr/local/Cellar/node@6/6.10.0/bin/node
  - /usr/local/bin/corber
  - serve
  - --environment=staging
  EXEC_PATH: /usr/local/Cellar/node@6/6.10.0/bin/node
  TMPDIR: /var/folders/3k/dyt4l2qs141cnldh6ptydmq40000gn/T
  SHELL: /bin/bash
  PATH:
  - /usr/local/opt/node@6/bin
  - /usr/local/sbin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /Users/AK/.composer/vendor/bin
  - /Applications/MySQLWorkbench.app/Contents/MacOS
  PLATFORM: darwin x64
  FREEMEM: 235675648
  TOTALMEM: 17179869184
  UPTIME: 250194
  LOADAVG: 2.53271484375,2.37890625,2.3916015625
  CPUS:
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz - 2700
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.10.1-DEV
  - http_parser: 2.7.0
  - icu: 58.2
  - modules: 48
  - node: 6.10.0
  - openssl: 1.0.2k
  - uv: 1.9.1
  - v8: 5.1.281.93
  - zlib: 1.2.8

ERROR Summary:

  - broccoliBuilderErrorStack: [undefined]
  - codeFrame: [undefined]
  - errorMessage: Unknown browser undefined
  - errorType: [undefined]
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
  - message: Unknown browser undefined
  - name: BrowserslistError
  - nodeAnnotation: [undefined]
  - nodeName: [undefined]
  - originalErrorMessage: [undefined]
  - stack: BrowserslistError: Unknown browser undefined
    at checkName (/Users/AK/Web/spotmii/customer-client/node_modules/browserslist/index.js:112:20)
    at Function.select (/Users/AK/Web/spotmii/customer-client/node_modules/browserslist/index.js:396:18)
    at /Users/AK/Web/spotmii/customer-client/node_modules/browserslist/index.js:140:33
    at Array.reduce (native)
    at resolve (/Users/AK/Web/spotmii/customer-client/node_modules/browserslist/index.js:121:18)
    at browserslist (/Users/AK/Web/spotmii/customer-client/node_modules/browserslist/index.js:217:16)
    at Class.included (/Users/AK/Web/spotmii/customer-client/node_modules/ember-weakmap/index.js:38:24)
    at addons.reduce (/usr/local/lib/node_modules/corber/node_modules/ember-cli/lib/models/addon.js:405:26)
    at Array.reduce (native)
    at Class.eachAddonInvoke (/usr/local/lib/node_modules/corber/node_modules/ember-cli/lib/models/addon.js:402:24)
```

**Comment by alexblom** [comment]
Thanks for this. I managed to get a replication case from the ember slack, hopefully a solution this week. It appears to be happening with the _newest_ version of ember-cli.

**Comment by OakBehringer** [comment]
Right on. Thanks for the reply. I'm also getting this when I hit `corber build`

> Corber: Path must be a string. Received undefined

Downgrading ember-cli to 3.1.4 fixed that one.

**Comment by OakBehringer** [comment]
Bummer! This still happens in ember-cli 3.1.4. (Unknown Browser)

**Comment by alexblom** [comment]
Thanks for the updates. I can confirm rolling back to 3.1.4 with / without the optimized `targets.js` is working, but not on later versions. For now that is your best solution.

I'm still working on a clean solution and should have a update end of week. We're considering a fast fix while scoping a more permanent solution which ultimately addresses #403 as well.

**Comment by alexblom** [comment]
The latest version (1.2.10) includes a fix for this, but only for ember-cli 3.1.4 and earlier. 

We’re still working on a fix for later versions of ember-cli and hope to have it out this week. Until then, keeping your ember-cli locked at 3.1.4 or earlier is the best solution.

**Comment by OakBehringer** [comment]
To be clear - upgrading to 1.2.10 didn't fix it for me, until I deleted the file "targets.js"

Is that what was expected?

**Comment by adentranter** [comment]
Currently on ember-cli 2.18.0 
I had this error happening on corber 1.2.9 , upgraded to 1.2.10 and its working for me now. 

Cheers @alexblom

**Comment by alexblom** [comment]
@adentranter No problems

@OakBehringer That was not expected, but we're still fully fixing the issues for `start` on later versios of ember-cli projects.

**Comment by alexblom** [comment]
**There is a temporary fix you can apply as of corber 1.2.10 for projects with an ember-cli version greater than 3.1.4**. Projects with pre 3.1.4 should not require any changes. We're still working on a more robust fix that will not impact corber projects running on earlier ember-cli versions. 

From your projects root directory:
```
cd node_modules/corber
vi package.json
#edit the ember-cli version to your projects version, e.g. 3.2.0
yarn / npm install
```

**Comment by alexblom** [comment]
There is a complete fix in #518. Some cleanup / CLI improvements to do before merging but you can expect it on Friday.

**Comment by alexblom** [comment]
Should be fully resolved as of 1.3.0. See the [release notes](https://github.com/isleofcode/corber/releases/tag/1.3.0) for details.
